modules/extensions-support/src/test/java/org/mule/module/extension/HeisenbergOperations.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.api.NestedProcessor;
+import org.mule.api.transport.PropertyScope;
+import org.mule.extension.ExtensionManager;
+import org.mule.extension.annotations.ImplementationOf;
+import org.mule.extension.annotations.Operation;
+import org.mule.extension.annotations.ParameterGroup;
+import org.mule.extension.annotations.RestrictedTo;
+import org.mule.extension.annotations.param.Optional;
+import org.mule.extension.annotations.param.Payload;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+@ImplementationOf(HeisenbergExtension.class)
+public class HeisenbergOperations
+{
+
+    private static final String SECRET_PACKAGE = "secretPackage";
+    private static final String METH = "meth";
+
+    private final HeisenbergExtension config;
+
+    public HeisenbergOperations(HeisenbergExtension config)
+    {
+        this.config = config;
+    }
+
+    @Inject
+    private ExtensionManager extensionManager;
+
+    @Operation
+    public String sayMyName()
+    {
+        return config.getPersonalInfo().getName();
+    }
+
+    @Operation
+    public void die()
+    {
+        config.setEndingHealth(HealthStatus.DEAD);
+    }
+
+    @Operation
+    public String getEnemy(int index)
+    {
+        return config.getEnemies().get(index);
+    }
+
+    @Operation
+    public String kill(@Payload String victim, String goodbyeMessage) throws Exception
+    {
+        return killWithCustomMessage(victim, goodbyeMessage);
+    }
+
+    @Operation
+    public String killWithCustomMessage(@Optional(defaultValue = "#[payload]") String victim, String goodbyeMessage)
+    {
+        return String.format("%s, %s", goodbyeMessage, victim);
+    }
+
+    @Operation
+    public String killMany(@RestrictedTo(HeisenbergExtension.class) List<NestedProcessor> killOperations, String reason) throws Exception
```

Comment:
Can you explain what @RestrictedTo annotation does? NestedProcessor is a mule API interface, why is needed in an extension implementation?